web/packages/design/src/SlideTabs/SlideTabs.tsx

```diff
@@ -112,77 +159,160 @@ type props = {
    * If true, disables pointer events.
    */
   disabled?: boolean;
+  /**
+   * If true, the control doesn't take as much horizontal space as possible,
+   * but instead wraps its contents.
+   */
+  fitContent?: boolean;
 };
 
-export type TabComponent = {
-  name: string;
-  component: React.ReactNode;
-};
+/**
+ * Definition of a tab. If it's a string, it denotes a name displayed on the
+ * tab. It's recommended to use a full object with tab panel ID for better
+ * accessibility.
+ */
+type TabSpec =
+  | string
+  | {
+      /** ID of the tab, unique in scope of this component. */
+      id: string;
+      /** Name displayed on the tab. */
+      content: ReactNode;
```

Comment:
We will get back to an arbitrary node anyway as soon as someone will invent an UI where part of it will use italics or whatever else can be customized (I did this more than one time), so I'd still argue that supporting arbitrary content costs us nothing. However, the case of icon sizes in particular is a valid point. That's something that we'll need to do anyway for icon buttons, since we don't have these standardized. @gzdunek Would it be OK to add special support for icons, but also support arbitrary content?

Reply:
>We will get back to an arbitrary node anyway as soon as someone will invent an UI where part of it will use italics or whatever else can be customized

We can always customize the appearance using a specific prop. For example, having a prop like `italics` in your case seems perfectly reasonable.
That said, are we sure we’ll need arbitrary components here? Their main strength is enabling highly customizable content, but it seems like we’ll primarily be dealing with text, perhaps with minor customizations.

> I'd still argue that supporting arbitrary content costs us nothing

I can think of a situation where someone passed a `Text` component with larger font size or something, and now I need to slightly change size of the `SlideTabs` component. Will that callsite break? I don't know it anymore, so I have to go and check it manually. So I think it is more difficult to maintain in the future.

As a side note, components that expect only simple props are easier to use. Every time I add a notification (mentioned in the Rafał's quote), I appreciate [how simple it its](https://github.com/gravitational/teleport/blob/4464ef93cf165c7723490ba28dcb0b87ccdeab79/web/packages/shared/components/Notification/types.ts#L38) comparing to my initial idea of allowing JSX there. I need a list or a button? I pass an array of strings or an object and the `Notification` component will take care of displaying them.

>Would it be OK to add special support for icons, but also support arbitrary content?

That’s an improvement, but as I mentioned earlier, do we really need arbitrary content here? It’s not a dealbreaker for me, but I'm thinking if we can design a simpler API.

Reply:
Alright then, string it is.